espressif/vscode-esp-idf-extension
# Enhance/whitespaces2

## Description

Enable usage of white spaces in project's path, esp idf's path and tools's path.
https://jira.espressif.com:8443/browse/VSC-808

Fixes #[1162](https://github.com/espressif/vscode-esp-idf-extension/issues/1162)
Fixes #[939](https://github.com/espressif/vscode-esp-idf-extension/issues/939)

## Type of change

Please delete options that are not relevant.
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Configure extension with whitespaces in esp idf's path, in the tools' path and project's path.
2. Test all features (building, flashing, monitoring etc)

All features should work, however, there is an issue currently with esp-idf while building a project with whitespaces:
```
 Run Build Command(s):C:/ESPRES~2/tools/ninja/111~1.1/ninja.exe cmTC_c5154 && [1/2] Building C object CMakeFiles/cmTC_c5154.dir/testCCompiler.c.obj
    FAILED: CMakeFiles/cmTC_c5154.dir/testCCompiler.c.obj 
    C:\ESPRES~2\tools\XTENSA~2\ESP-13~1.0_2\XTENSA~1\bin\XT8994~1.EXE   -mlongcalls -Wno-frame-address -o CMakeFiles/cmTC_c5154.dir/testCCompiler.c.obj -c "C:/Users/Radu/Desktop/esp projects/esp     spaces/blink/build/CMakeFiles/CMakeTmp/testCCompiler.c"
    thread 'main' panicked at 'assertion failed: `(left == right)`
      left: `"XT8994~1.EXE"`,
     right: `"xtensa"`: Called tool must have pattern "xtensa-esp*-elf-*"', main.rs:46:18
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    ninja: build stopped: subcommand failed.
```

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Tested as described above on Windows 11 with different shells, including: Command Prompt, Powershell and Bash

**Test Configuration**:
* ESP-IDF Version: 5.2.1
* OS (Windows,Linux and macOS): Windows

## Checklist
